plugin.rb

```diff
@@ -27,22 +30,155 @@ class Engine < ::Rails::Engine
       engine_name PLUGIN_NAME
       isolate_namespace Salesforce
     end
+
+    %i{
+      leads
+      contacts
+    }.each do |type|
+      define_singleton_method("#{type}_group") do
```

Comment:
Since we only have to define two methods, I think it is much better for readability if we just define the two methods manually. Also, it makes it easier to search for where the methods are defined.